components/console_cmd_mqtt/examples/mqtt-basic/main/mqtt-basic.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Unlicense OR CC0-1.0
+ */
+#include <stdio.h>
+#include "esp_netif.h"
+#include "nvs_flash.h"
+#include "esp_netif.h"
+#include "esp_event.h"
+#include "console_mqtt.h"
+
+
+void app_main(void)
+{
+    ESP_ERROR_CHECK(esp_netif_init());
+    ESP_ERROR_CHECK(esp_event_loop_create_default());
+    esp_err_t ret = nvs_flash_init();   //Initialize NVS
+    if (ret == ESP_ERR_NVS_NO_FREE_PAGES || ret == ESP_ERR_NVS_NEW_VERSION_FOUND) {
+        ESP_ERROR_CHECK(nvs_flash_erase());
+        ret = nvs_flash_init();
+    }
+    ESP_ERROR_CHECK(ret);
+
+    // Initialize console REPL
+    ESP_ERROR_CHECK(console_cmd_init());
+
+    // Register mqtt command
+    ESP_ERROR_CHECK(console_cmd_mqtt_register());
+
+    // start console REPL
+    ESP_ERROR_CHECK(console_cmd_start());
```

Comment:
If I read the code correctly, this example implements no network connectivity, so all mqtt commands would just fail.

(I think you can add mqtt or ethernet component or simply add the connect component)

Reply:
Add common_connect for connection